test/service.go

```diff
@@ -65,9 +65,41 @@ func CreateService(ctx *Context, name, namespace, image string) (*servingv1beta1
 	return service, nil
 }
 
+func GetPod(ctx *Context, ns string) {
+	// run infinite for loop because once after the Global proxy update operator pod will be recreated
+	// and which intern updates knative serving controller pod so we don't no exact time duration for this to happen
+	for {
+		pods, _ := ctx.Clients.Kube.CoreV1().Pods(ns).List(metav1.ListOptions{
+			LabelSelector: "app=controller",
+		})
+
+		for i := range pods.Items {
+			for _, container := range pods.Items[i].Spec.Containers {
+				for _, e := range container.Env {
+					if e.Name == "HTTP_PROXY" && e.Value != "" {
+						if pods.Items[i].Status.Phase == corev1.PodRunning {
```

Comment:
This doesn't suffice to make sure the pod is actually running I think. I had lots of fun doing this for a separate upstream bug, this is what I came up with:

```golang

func isPodReady(pod *corev1.Pod) bool {
	if pod.DeletionTimestamp != nil || pod.Status.PodIP == "" {
		return false
	}
	for _, cond := range pod.Status.Conditions {
		if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
			return true
		}
	}
	return false
}
```

Reply:
I'd also check if the pod is ready first, before even looping through the containers and looking for the proxy values.

Reply:
even if we check pod is ready first we need to validate whether that pod have env or not
So will that make any impact if we add check for podReady at the first step?

Reply:
It won't, it just reads strange to me to check the env first and then check for readiness.

Reply:
Why i am checking env because some time pod will not restart and pod will be always running in that case assuming Pod Ready will miss lead i feel